packages/dashboard/src/components/delivery-alert-store.tsx

```diff
@@ -0,0 +1,539 @@
+  const rmf = React.useContext(RmfAppContext);
+  const [alerts, setAlerts] = React.useState<Record<string, DeliveryAlertData>>({});
+  const appController = React.useContext(AppControllerContext);
+
+  const filterAndPushDeliveryAlert = (deliveryAlert: DeliveryAlert, taskState?: TaskState) => {
+    // Check if a delivery alert for a task is already open, if so, replace it
+    // with this new incoming deliveryAlert.
+    setAlerts((prev) => {
+      if (!deliveryAlert.task_id) {
+        return {
+          ...prev,
+          [deliveryAlert.id]: { deliveryAlert, taskState },
+        };
+      }
+
+      // TODO(ac): set action to cancelled for delivery alerts that have been
+      // updated.
+      const filteredAlerts = Object.fromEntries(
+        Object.entries(prev).filter(
+          ([_, alertData]) =>
+            !alertData.deliveryAlert.task_id ||
+            alertData.deliveryAlert.task_id !== deliveryAlert.task_id,
+        ),
+      );
+      filteredAlerts[deliveryAlert.id] = {
+        deliveryAlert,
+        taskState,
+      };
+      return filteredAlerts;
+    });
+  };
+
+  React.useEffect(() => {
+    if (!rmf) {
+      return;
+    }
+
+    // Initialize with any existing delivery alerts that are still waiting for
+    // action.
+    (async () => {
+      // TODO(ac): Create an endpoint which only gives delivery alerts that have
+      // not been actioned upon, to prevent requesting too many delivery alerts.
+      let deliveryAlerts: DeliveryAlert[] = [];
+      try {
+        deliveryAlerts = (await rmf.deliveryAlertsApi.getDeliveryAlertsDeliveryAlertsGet()).data;
+      } catch (e) {
+        console.error(`Failed to retrieve existing delivery alerts: ${e}`);
+        return;
+      }
+
+      const filteredAlertsMap: Record<string, DeliveryAlertData> = {};
+      const taskIdToAlertsMap: Record<string, DeliveryAlertData> = {};
+      const resolvedTaskIds: string[] = [];
+      for (const alert of deliveryAlerts) {
+        if (alert.action !== 'waiting') {
+          // Keeping track of resolved task delivery alerts to filter any
+          // remaining warning and waiting delivery alerts.
+          if (alert.task_id && alert.tier === 'error') {
+            resolvedTaskIds.push(alert.task_id);
+          }
+          continue;
+        }
+
+        // No task involved, and still waiting for user action. There should not
+        // be any longstanding delivery alerts that appear after a refresh, only
+        // the delivery alerts that are currently present and have not been
+        // responded to.
+        if (!alert.task_id) {
+          filteredAlertsMap[alert.id] = { deliveryAlert: alert, taskState: undefined };
+          continue;
+        }
+
+        // Tasks that have not been encountered before will go into
+        // taskIdToAlertsMap first, as an error will supercede a warning.
+        if (!Object.keys(taskIdToAlertsMap).includes(alert.task_id)) {
```

Comment:
Since it is an if-else, the variable has a local scope, so I think we can use const and add it in the try. That makes the code readable because it assumes that the variable will be from the state of the api response. 

Also, I think the else has to follow the same structure and add the logic inside the try and not outside. 
```suggestion
if (!Object.keys(taskIdToAlertsMap).includes(alert.task_id)) {
  try {
    const state = (await rmf.tasksApi.getTaskStateTasksTaskIdStateGet(alert.task_id)).data;

    taskIdToAlertsMap[alert.task_id] = {
      deliveryAlert: alert,
      taskState: state,
    };
  } catch (error) {
    console.error(`Failed to fetch task state for ${alert.task_id} for delivery alert ${alert.id}`);
  }
} else if (alert.tier === 'error') {
  try {
    const state = (await rmf.tasksApi.getTaskStateTasksTaskIdStateGet(alert.task_id)).data;

    taskIdToAlertsMap[alert.task_id] = {
      deliveryAlert: alert,
      taskState: state,
    };
  } catch (error) {
    console.error(`Failed to fetch task state for ${alert.task_id} for delivery alert ${alert.id}`);
  }
}
```

Reply:
Besides, since we are doing 
```
taskIdToAlertsMap[alert.task_id] = {
            deliveryAlert: alert,
            taskState: state,
          };

```
Maybe in the if statement we can do something like

`if (!taskIdToAlertsMap[alert.task_id]) `

Reply:
Changed if statement in https://github.com/open-rmf/rmf-web/pull/768/commits/83666bba825ec114cc679179300587c135db958a